cpp/src/arrow/dataset/file_csv_test.cc

```diff
@@ -199,18 +210,11 @@ TEST_P(TestCsvFileFormat, OpenFailureWithRelevantError) {
   if (GetCompression() != Compression::type::UNCOMPRESSED) {
     GTEST_SKIP() << "File source name is different with compression";
   }
-  auto source = GetFileSource("");
-  EXPECT_RAISES_WITH_MESSAGE_THAT(Invalid, testing::HasSubstr("<Buffer>"),
-                                  format_->Inspect(*source).status());
-
-  constexpr auto file_name = "herp/derp";
-  ASSERT_OK_AND_ASSIGN(
-      auto fs, fs::internal::MockFileSystem::Make(fs::kNoTime, {fs::File(file_name)}));
-  EXPECT_RAISES_WITH_MESSAGE_THAT(Invalid, testing::HasSubstr(file_name),
-                                  format_->Inspect({file_name, fs}).status());
+  TestOpenFailureWithRelevantError(format_.get(), StatusCode::Invalid);
```

Comment:
If you add a `GetFormat` to the `WriterMixin` you could probably save a lot of `format_.get()` arguments.